File: Codeforces/Golang/800/236A - Boy or Girl/main.go
package main

import "fmt"

func main() {
	var name string
	fmt.Scan(&name)

	distinctCh := make(map[rune]bool)

	for _, char := range name {
		distinctCh[char] = true
	}
	count := len(distinctCh)

	if count%2 == 0 {
		fmt.Println("CHAT WITH HER!")
	} else {
		fmt.Println("IGNORE HIM!")
	}
}

File: Codeforces/Rust/800/236A - Boy or Girl/src/main.rs
use std::collections::HashSet;
use std::io;

fn main() {
    let mut input = String::new();
    let _ = io::stdin().read_line(&mut input);

    let name = input.trim();

    let names: HashSet<char> = name.chars().collect();

    if names.len() % 2 == 0 {
        println!("CHAT WITH HER!")
    } else {
        println!("IGNORE HIM!")
    }
}
